predicators/envs/doors.py

```diff
@@ -729,3 +729,447 @@ def _get_open_door_target_value(mass: float, friction: float,
         # A made up complicated function.
         return np.tanh(target_rot) * (np.sin(mass) +
                                       np.cos(friction) * np.sqrt(mass))
+
+
+class DoorKnobsEnv(DoorsEnv):
+    """A 2D navigation environment with obstacles, rooms, and doors."""
+    # Types
+    _robot_type = Type("robot", ["x", "y"])
+    _door_type = Type("door", ["x", "y", "theta", "mass", "friction", "open"])
+    _knob_type = Type("knob", ["x", "y", "theta", "rot", "target_rot", "open"])
+    _room_type = Type("room", ["x", "y"])
+    _obstacle_type = Type("obstacle", ["x", "y", "width", "height", "theta"])
+    open_door_thresh: ClassVar[float] = 0.1
```

Comment:
looks like a significant amount of this is copied from the `DoorsEnv` superclass above. We want to minimize code duplication - especially when subclassing - so please only override/change the types that are different from the superclass